amqp/src/main/java/io/cloudevents/amqp/ProtonAmqpMessageFactory.java

```diff
@@ -41,62 +44,63 @@ private ProtonAmqpMessageFactory() {
     }
 
     /**
-     * Creates a MessageReader to read a proton-based {@link Message}.
+     * Creates a {@link MessageReader} to read a proton-based {@link Message}.
+     * This reader is able to read both binary and structured encoded {@link io.cloudevents.CloudEvent}.
      * <p>
      * This implementation simply calls {@link #createReader(String, ApplicationProperties, byte[])}.
      *
-     * @param message The proton message to read from.
-     *
-     * @return        A message reader that can read the given proton message to a cloud event representation.
+     * @param message The proton {@link Message} to read from.
+     * @return A {@link MessageReader} that can read the given proton {@link Message} to a {@link io.cloudevents.CloudEvent} representation.
+     * @throws CloudEventRWException if something goes wrong while resolving the {@link SpecVersion} or if the message has unknown encoding
+     * @see #createReader(String, ApplicationProperties, byte[])
      */
-    public static MessageReader createReader(final Message message) {
-
+    public static MessageReader createReader(final Message message) throws CloudEventRWException {
         final byte[] payload = AmqpConstants.getPayloadAsByteArray(message.getBody());
         return createReader(message.getContentType(), message.getApplicationProperties(), payload);
     }
 
     /**
-     * Creates a MessageReader using the content-type property and payload of a proton-based message.
-     * <p>
-     * This method simply calls {@link #createReader(String, ApplicationProperties, byte[])}.
+     * Creates a {@link MessageReader} using the {@code content-type} property and payload of a proton-based {@link Message}.
```

Comment:
@slinkydeveloper: do you think it will help the users if we mention here that this method will  create a structured reader? With the binary reader, the application-properties is mandatory but it is  optional for a structured reader

Reply:
I think we should not have at all this method TBH, since the user doesn't usually know if she/he's reading binary or structured mode events.

Reply:
i was thinking the same.. then how about exposing a single `createReader` API that accepts a proton `Section body` instead of `byte[]`? i.e
```Java
createReader(final String contentType, final ApplicationProperties props, @Nullable final Section payload)
```
Then the factory class can internally convert the body Section to `byte[]`, saving the user from doing so before using the API. WDYT?

Reply:
Let me try to do it

Reply:
How does it looks like now?

Reply:
perfect! LGTM